### PR TITLE
doc/reference/storage/settings: fix misleading examples

### LINKED ISSUE #15451: doc: settings Misleading examples
In both examples function `foo_settings_set` is used as `.h_set` callback. There is `settings_val_read_cb(value_ctx, &foo_val, sizeof(foo_val));` used as return value. The function typically returns number of bytes read, which is treated as error code by callback (when there is no error, zero should be returned).
This makes documentation misleading for reader.

### PR DESCRIPTION
**Pull request by nvlsianpu**
Code snippets show misleading example of read_callback usage
in h_set handlers.

This patch aligns snippets to implemented behavior.

fixes #15451

Signed-off-by: Andrzej Puzdrowski <andrzej.puzdrowski@nordicsemi.no>

### CODE REVIEW

**Comment by nvlsianpu** [comment]
cc @MarekPieta @kapi-no

File: doc/reference/storage/settings/settings.rst [hunk 1]
```diff
@@ -141,9 +142,27 @@ export functionality, for example, writing to the shell console).
...
     {
+        int rc;
+
         if (argc == 1) {
             if (!strcmp(argv[0], "bar")) {
-                return read_cb(cb_arg, &foo_val, sizeof(foo_val));
+                rc = read_cb(cb_arg, &foo_val, sizeof(foo_val));
+                if (rc >= 0) {
+                    /* key-value pair was properly read.
+                     * rc contins value length.
```
Comment (kapi-no) [inline doc/reference/storage/settings/settings.rst]: contains

File: doc/reference/storage/settings/settings.rst [hunk 2]
```diff
@@ -181,9 +200,14 @@ up from where it was before restart.
...
     {
+        int rc;
+
         if (argc == 1) {
             if (!strcmp(argv[0], "bar")) {
-                return read_cb(cb_arg, &foo_val, sizeof(foo_val));
+                rc = read_cb(cb_arg, &foo_val, sizeof(foo_val));
+                if (rc >= 0) {
+                    return 0;
+                }
```
Comment (kapi-no) [inline doc/reference/storage/settings/settings.rst]: Shouldn`t we return rc when there is an error?

```suggestion
                } else {
                    return rc;
                }
```

File: doc/reference/storage/settings/settings.rst [hunk 3]
```diff
@@ -141,9 +142,27 @@ export functionality, for example, writing to the shell console).
...
-                return read_cb(cb_arg, &foo_val, sizeof(foo_val));
+                rc = read_cb(cb_arg, &foo_val, sizeof(foo_val));
+                if (rc >= 0) {
+                    /* key-value pair was properly read.
+                     * rc contins value length.
+                     * key-value is deleted if length equals to 0.
+                     * Let's return success.
+                     */
+                    if (rc == 0) {
+                        /* assign something to the value as its
```
Comment (kapi-no) [inline doc/reference/storage/settings/settings.rst]: IMO, this sounds a bit better:
set the default value as its...

**Review by kapi-no — CHANGES_REQUESTED** [review]

**Comment by ioannisg** [comment]
@dbkinder could you review this one, so we can get it in for 2.0.0 release?

File: doc/reference/storage/settings/settings.rst [hunk 4]
```diff
@@ -141,9 +143,25 @@ export functionality, for example, writing to the shell console).
...
+        int rc;
+
         if (argc == 1) {
             if (!strcmp(argv[0], "bar")) {
-                return read_cb(cb_arg, &foo_val, sizeof(foo_val));
+                rc = read_cb(cb_arg, &foo_val, sizeof(foo_val));
+                if (rc >= 0) {
+                    /* key-value pair was properly read.
+                     * rc contains value length.
+                     * key-value is deleted if length equals to 0.
```
Comment (ioannisg) [inline doc/reference/storage/settings/settings.rst]: equals 0 (remove "to")

**Review by ioannisg — COMMENTED** [review]
A minor suggestion.

File: doc/reference/storage/settings/settings.rst (hunk 4 above)
Comment (nvlsianpu) [reply doc/reference/storage/settings/settings.rst]: fixed

**Review by ioannisg — APPROVED** [review]

**Review by kapi-no — APPROVED** [review]

**Review by MarekPieta — APPROVED** [review]

**Review by dbkinder — APPROVED** [review]
doc change LGTM.

**Pull request merged**